### PR TITLE
Document Modelling Refactor V3

### PR DESCRIPTION
**Pull request by logan-markewich**
# Description

Massive change, appologies in advance.

TLDR: see the llama_index/schema.py file for the new node schemas.

## Type of Change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## TODO

- [x] appease the mypy gods
- [x] make sure unit tests pass
- [x] re-think node-serialization for vector stores that store text
- [x] clean up references to `doc_id` and `extra_info`
- [x] write conversion script for legacy indexes
- [x] update docs 
- [x] fully test changes (ALL vector stores, as many notebooks as possible)
- [ ] one last merge of main before merging PR

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+
+### New Features
+- New `TextNode`/`Document` object classes based on pydantic (#6586)
+- `TextNode`/`Document` objects support metadata customization (metadata templates, exclude metadata from LLM or embeddings) (#6586)
+- Nodes no longer require flat metadata dictionaries, unless the vector store you use requires it (#6586)
+
+### Breaking/Deprecated API Changes
+- `Node` has been renamed to `TextNode` and is imported from `llama_index.schema` (#6586)
+- `TextNode` and `Document` must be instansiated with kwargs: `Document(text=text)` (#6586)
+- `TextNode` (fka `Node`) has a `id_` or `node_id` property, rather than `doc_id` (#6586)
```
Comment (Disiok) [inline CHANGELOG.md:13]: very nit: typo on `aka`?

Comment (logan-markewich) [reply CHANGELOG.md:13]: `fka` = formally known as (there is technically no object called `Node` now)
